rules/number-literal-case.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+const fix = value => {
+	if (!/^0[a-zA-Z]/.test(value)) {
+		return value;
+	}
+
+	const indicator = value[1].toLowerCase();
+	const val = value.substring(2).toUpperCase();
```

Comment:
I generally prefer `.slice()` when possible as it's more common. @jfmengels Maybe we should have a rule about that?